docs/design/rosbag2_snapshot_mode.md

```diff
@@ -0,0 +1,46 @@
+# Adding a Snapshot Service for rosbag2
+
+## Context
+In [ros2/rosbag2#663](https://github.com/ros2/rosbag2/issues/663), it was recommended that a "snapshot" feature be added to rosbag2.
+
+## Motivation
+The goal of this feature is to provide the user with a mechanism to maintain a transient cache of the recent messages sent on specified topics that is only written to permanent storage upon a user-invoked trigger.
+
+
+For example, if a rare, anomalous behavior occurs, this feature will allow the user to record the most recent ROS topic messages when the behavior is observed, instead of having to record the entire execution.
+
+## Example Usage
+
+Starting `rosbag2` in snapshot mode with a specified cache size:
+```
+$ ros2 bag record --max-cache-size 100000 --snapshot-mode [topics [topics ...]]
+```
+
+Triggering a snapshot:
```

Comment:
This is good, nitpick: I would probably shorten the language to `~/snapshot` (note `~` notation for "root namespace of node").

I think you should also make an additional note that this should also be usable via keyboard handler on the recording process. We don't have any keyboarding hooked up _yet_, but it should be an intention for the feature. Not necessary to tie it to a key, but we could say something like "Also triggerable via key press on recording process, suggestion: 's'"